components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -359,6 +359,7 @@ impl IndexScanExecutorImpl {
         value: &[u8],
         columns: &mut LazyBatchColumnVec,
     ) -> Result<()> {
+        let tail_len = value[0] as usize;
```

Comment:
The comment above says `tail_len` is always `0x00`. In what cases does the rule break?

Reply:
In the case tidb mistakenly write untouched key to tikv.

Reply:
@iosmanthus Could you update comments for that case?

Reply:
OK